// Copyright 2019 gRPC authors.

syntax = "proto3";

package demo;

message Request {
    int64 client_id = 1;
    string request_data = 2;
}

message Response {
    int64 server_id = 1;
    string response_data = 2;
}

service GRPCDemo {

    rpc SimpleMethod (Request) returns (Response);

    rpc ClientStreamingMethod (stream Request) returns (Response);

    rpc ServerStreamingMethod (Request) returns (stream Response);

    rpc BidirectionalStreamingMethod (stream Request) returns (stream Response);
}
